theseus/theseus_layer.py

```diff
@@ -242,12 +249,17 @@ def backward(ctx, *grad_outputs):
 class _DLMPerturbation(CostFunction):
     def __init__(
         self,
-        var: LieGroup,
+        var: Manifold,
         epsilon: Variable,
         grad: Variable,
         cost_weight: CostWeight,
         name: Optional[str] = None,
     ):
+        if not isinstance(var, LieGroup):
```

Comment:
LieGroup -> Manifold? following the function signature.

Reply:
No, we need this to be a `LieGroup` for the projection operator to be defined. Manifold is more general than needed, but avoids some annoying mypy errors when calling this function a level above.